data-access-layer/eventmesh/ApiServerClient.js

```diff
@@ -669,6 +756,246 @@ class ApiServerClient {
       })
       .then(resource => _.get(resource, 'spec.options.context'));
   }
+
+  /**
+   * @description Create OSB Resource in Apiserver with the opts
+   * @param {string} opts.resourceGroup - Name of resource group 
+   * @param {string} opts.resourceType - Type of resource 
+   * @param {string} opts.resourceId - Unique id of resource 
+   * @param {string} opts.metadata - Optional; pass finalizers or some other field
+   * @param {string} opts.labels - to be put in label
+   * @param {Object} opts.spec - Value to set for spec field of resource
+   * @param {string} opts.status - status of the resource
+   */
+  createOSBResource(opts) {
+    logger.info(`Creating resource with opts: `, opts);
+    assert.ok(opts.resourceGroup, `Property 'resourceGroup' is required to create resource`);
+    assert.ok(opts.resourceType, `Property 'resourceType' is required to create resource`);
+    assert.ok(opts.resourceId, `Property 'resourceId' is required to create resource`);
+    assert.ok(opts.spec, `Property 'spec' is required to create resource`);
+    const metadata = _.merge(opts.metadata, {
+      name: opts.resourceId
+    });
+    if (opts.labels) {
+      metadata.labels = opts.labels;
+    }
+    const crdJson = this.getCrdJson(opts.resourceGroup, opts.resourceType);
+    const resourceBody = {
+      apiVersion: `${crdJson.spec.group}/${crdJson.spec.version}`,
+      kind: crdJson.spec.names.kind,
+      metadata: metadata,
+      spec: camelcaseKeys(opts.spec)
```

Comment:
This is because the OSB attributes are of format with `-` however, kubebuilder and kubernetes attributes cannot have `-`. Hence they are camescased. 

The npm module does camelcasing of all the keys having `-`, not only of a string. Hence the lodash library and this one has different usecase.

Reply:
@jagadish-kb I have put a comment explaining why `camelcase-keys` module is being used.
About translating it back, as discussed whoever is fetching ServiceInstance crd already know what the structure of CRD is and mapping is also very intuitive. If this translating back code is repeated in future then we can write common `get` function in `ApiServerClient` class itself